config/default.config

```diff
@@ -30,6 +32,14 @@ params {
     docker_image_samtools = "${-> params.docker_container_registry}/samtools:${params.samtools_version}"
 
     gatk_ir_compression = 1
+
+    // These parameters are inferred from the input files. The user can override them in the config file if required.
+    reference_fasta_fai = "${-> params.reference_fasta}.fai"
+    reference_fasta_dict = "${-> Nextflow.file(params.reference_fasta).resolveSibling(Nextflow.file(params.reference_fasta).getBaseName() + '.dict')}"
```

Comment:
**question (non-blocking)**: Any reason to use `resolveSibling` for the dict but not for the other files?

Reply:
The other files append to the file extension (e.g. `reference.fasta.fai`) but the dictionary replaces it (e.g. `reference.dict`).

https://github.com/uclahs-cds/pipeline-recalibrate-BAM/blob/25a0b8e534de2addff748c744b84b34375248ab5/test/configtest-F16.json#L231-L233

Reply:
Ah right, it's to keep the full path intact since basename only gives the filename